march_hardware_interface/src/march_hardware_interface.cpp

```diff
@@ -191,6 +191,11 @@ void MarchHardwareInterface::validate()
     this->outsideLimitsCheck(i);
     this->iMotionCubeStateCheck(i);
```

Comment:
```suggestion
    if (this->iMotionCubeStateCheck(i)) {
      fault_state = true;
    }
```
Make sure to also declare `fault_state` before the loop (after line 188)

Reply:
Down here, you say that it should return `false` when something is wrong and `true` when everything's good, so you should invert this logic statement

Reply:
I don't understand the declaration of the variable after line 188. I thought variables need to be declared in the header file?

Reply:
variables declared in the class header are considered member variables and can be accessed through the `this` object. Local variables, like the one Roel suggests can be declared inside a function and will only be accessible inside the function. Since you only use this variable inside this function, it is better to declare it after line 188.